client/components/RequireAuth.tsx

```diff
@@ -2,18 +2,13 @@ import React, { PropsWithChildren } from 'react';
 import { Navigate, useLocation } from 'react-router-dom';
 import { useAuth } from '../auth/AuthProvider';
 
-interface HeaderProps {
-    children: any;
-}
-
-// function RequireAuth(): JSX.Element {
-// eslint-disable-next-line react/function-component-definition
-const RequireAuth: React.FC<PropsWithChildren<HeaderProps>> = ({ children }) => {
-    // <div>{children}</div>
-
-    const auth = useAuth();
+function RequireAuth({ children }: React.PropsWithChildren<unknown>) {
```

Comment:
`children` needs to be typed as a ReactElement otherwise an error is thrown in `index.tsx` when using the `RequireAuth` component. The following did the trick for me. 
```typescript
function RequireAuth({children}: { children: React.ReactElement | null; }): React.ReactElement | null {
  ...
}
````

Reply:
I played around with this for a while and what I ended up with may differ from whats here. I'm not getting any errors now but I know the ones your talking about.